chain/types/electionproof_test.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import (
+	"fmt"
+	"math/big"
+	"os"
+	"testing"
+)
+
+func TestElectionLam(t *testing.T) {
+	p := big.NewInt(64)
+	tot := big.NewInt(128)
+	lam := lambda(p, tot)
+	target := 64. * 5. / 128.
+	if fxToD(lam) != target {
+		t.Fatalf("wrong lambda: %f, should be: %f", fxToD(lam), target)
+	}
+}
+
+func TestElectionExp(t *testing.T) {
+	t.Skip()
+	const N = 256
+
+	step := big.NewInt(5)
+	step = step.Lsh(step, 256) // Q.256
+	step = step.Div(step, big.NewInt(N-1))
+
+	x := big.NewInt(0)
+	for i := 0; i < N; i++ {
+		y := expneg(x)
+		fmt.Printf("\"%s\" \"%s\";\n", x, y)
```

Comment:
Ideally, instead of just printing these for manual inspection, you could programmatically compare the first k bits to ground-truth, as [here](https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/reward/reward_state_test.go). I have the sense you've done enough testing already, but it's just in case others come along later and modify the code somehow and say "well, I don't know what to make of this output, but the tests still pass and are clearly doing something, seems fine"

Reply:
I've done manual tests but the plan is to add more automatic ones.